core/capstanignore.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package core
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type Capstanignore interface {
+	LoadFile(path string) error
+	AddPattern(pattern string) error
+	PrintPatterns()
+	IsIgnored(path string) bool
+}
+
+var CAPSTANIGNORE_ALWAYS []string = []string{"/meta/*", "/mpm-pkg/*", "/.git/*"}
+
+// CapstanignoreInit creates a new capstanignore struct that is
```

Comment:
I agree, interface name should be mentioned here, not the struct name. I think, however, that the interface name should remain spelled as `Capstanignore` and not `CapstanIgnore` as capstanignore is considered a single word IMO. Would you prefer to have it CamelCase anyway?